tools/DevDiagnostics/DevHome.DevDiagnostics/Models/SensorNode.cs

```diff
@@ -0,0 +1,294 @@
+                                            }
+                                            else if (_sensor.Value < 1048576)
+                                            {
+                                                result = CommonHelper.GetLocalizedString("ConnectionSpeedSensorFormatKbps", _sensor.Value / 1024);
+                                            }
+                                            else if (_sensor.Value < 1073741824)
+                                            {
+                                                result = CommonHelper.GetLocalizedString("ConnectionSpeedSensorFormatMbps", _sensor.Value / 1048576);
+                                            }
+                                            else
+                                            {
+                                                result = CommonHelper.GetLocalizedString("ConnectionSpeedSensorFormatGbps", _sensor.Value / 1073741824);
+                                            }
+
+                                            break;
+                                    }
+
+                                    break;
+                                }
+
+                            default:
+                                {
+                                    const int _1MB = 1048576;
+                                    result = _sensor.Value < _1MB ? CommonHelper.GetLocalizedString("ConnectionSpeedSensorFormatKbps", _sensor.Value / 1024) : CommonHelper.GetLocalizedString("ConnectionSpeedSensorFormatMbps", _sensor.Value / _1MB);
+                                    break;
+                                }
+                        }
+
+                        return result;
+                    }
+
+                case SensorType.TimeSpan:
+                    {
+                        return CommonHelper.GetLocalizedString(_unitFormat, TimeSpan.FromSeconds(_sensor.Value.Value));
+                    }
+
+                default:
+                    {
+                        return CommonHelper.GetLocalizedString(_unitFormat, _sensor.Value);
+                    }
+            }
+        }
+
+        return string.Empty;
+    }
+}
+
+public interface IHardwareMonitorTreeNode
+{
+    string Name
+    {
+        get; set;
+    }
+}
+
+public partial class HardwareNode : ObservableObject, IHardwareMonitorTreeNode
+{
+    public HardwareNode(string name)
+    {
+        _name = name;
+    }
+
+    [ObservableProperty]
+    private string _name;
+
+    [ObservableProperty]
+    private ObservableCollection<IHardwareMonitorTreeNode> _children = new();
+}
+
+public partial class SensorGroupNode : ObservableObject, IHardwareMonitorTreeNode
+{
+    public SensorGroupNode(SensorType type, List<SensorNode> sensors)
+    {
+        _children = new ObservableCollection<SensorNode>(sensors);
+
+        switch (type)
+        {
+            case SensorType.Voltage:
+                _name = CommonHelper.GetLocalizedString("SensorTypeVoltage");

```

Comment:
As above, perhaps consider a map of some kind?

Reply:
Same as above, will keep the switch statement since it's efficient this way